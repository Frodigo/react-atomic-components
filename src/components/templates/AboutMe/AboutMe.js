import React from 'react';

import PageHeader from '../../organisms/PageHeader';
import Newsletter from '../../organisms/Newsletter';
import PageFooter from '../../organisms/PageFooter/PageFooter';
import ShareSection from '../../organisms/ShareSection';
import MobileNav from '../../organisms/MobileNav/MobileNav';
import { items as mobileNavItems } from '../../organisms/MobileNav/MobileNav.config';
import Heading from '../../atoms/Text/Heading';
import MediaBlockSmaller from '../../molecues/MediaBlock/MediaBlockSmaller';
import ImageCarousel from '../../organisms/ImageCarousel';
import Paragraph from '../../atoms/Text/Paragraph';
import Blockquote from '../../atoms/Text/Blockquote';
import Slider from '../../organisms/Slider';
import ContentBlock from '../../molecues/ContentBlock';

import classes from './AboutMe.module.css';
import FooterMock from "../partials/FooterMock";

const Home = (props) => {
    const { ...rest } = props;

    const image = null;

    const imageCarouselProps = {
        items: [
            {
                url: 'https://via.placeholder.com/208x117',
                alt: 'figure'
            },
            {
                url: 'https://via.placeholder.com/260x117',
                alt: 'figure'
            },
            {
                url: 'https://via.placeholder.com/108x117',
                alt: 'figure'
            },
            {
                url: 'https://via.placeholder.com/58x117',
                alt: 'figure'
            }
        ],
        title: 'Tech skills'
    };

    const testimonialsItems = [
        <Blockquote
            key="1"
            author="Author 1"
            quote="Lorem ipsum"
            type="inverse"
        />,
        <Blockquote
            key="2"
            author="Author 2"
            quote="Lorem ipsum"
            type="inverse"
        />,
        <Blockquote
            key="3"
            author="Author 3"
            quote="Lorem ipsum"
            type="inverse"
        />
    ];

    const pageHeaderNavItems = [
        {
            url: '#',
            label: 'Home'
        },
        {
            url: '#',
            label: 'About me',
            isActive: true
        },
        {
            url: '#',
            label: 'Blog'
        },
        {
            url: '#',
            label: 'Get in touch'
        }
    ];

    return (
        <div className={classes.about} {...rest}>
            <PageHeader navItems={pageHeaderNavItems} />
            <div className={classes.content}>
                <figure
                    className={classes.mainFigureDesktop}
                    style={{
                        backgroundImage: `url("${image}")`
                    }}
                />
                <section className={classes.mainContent}>
                    <Heading type="h1">Hey!</Heading>
                    <figure
                        className={classes.mainFigure}
                        style={{
                            backgroundImage: `url("${image}")`
                        }}
                    />
                    <p>
                        Nam porttitor blandit accumsan. Ut vel dictum sem, a
                        pretium dui. In malesuada enim in dolor euismod, id
                        commodo mi consectetur. Curabitur at vestibulum nisi.
                        Nullam vehicula nisi velit. Mauris turpis nisl, molestie
                        ut ipsum et, suscipit vehicula odio. Vestibulum interdum
                        vestibulum felis ac molestie.
                    </p>
                </section>
            </div>
            <section className={classes.experience}>
                <Heading type="h2">Work experience</Heading>

                <MediaBlockSmaller
                    imageUrl="https://via.placeholder.com/240x240"
                    subtitle="05/2016 - 04/2020"
                    title="Frontend Tech Leader"
                    text="Divante"
                />

                <MediaBlockSmaller
                    imageUrl="https://via.placeholder.com/240x240"
                    subtitle="05/2016 - 04/2020"
                    title="Frontend Tech Leader"
                    text="Divante"
                />
            </section>
            <div className={classes.skills}>
                <ImageCarousel {...imageCarouselProps} />
            </div>
            <div className={classes.certifications}>
                <Heading type="h2">Certifications</Heading>
                <Paragraph type="large">
                    Lorem ipsum dolor sit amet Lorem ipsum dolor sit amet Lorem
                    ipsum dolor sit amet Lorem ipsum dolor sit amet Lorem ipsum
                    dolor sit amet
                </Paragraph>

                <ul className={classes.certificationList}>
                    <li>
                        {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
                        <a href="#">Certification 1</a>
                        (10/2020)
                    </li>
                    <li>
                        {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
                        <a href="#">Certification 2</a>
                        (10/2020)
                    </li>
                    <li>
                        {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
                        <a href="#">Certification 3</a>
                        (10/2020)
                    </li>
                </ul>
            </div>
            <div className={classes.testimonials}>
                <Heading type="h2" classes={classes.testimonialsHeading}>
                    Testimonials
                </Heading>
                <Slider items={testimonialsItems} />
            </div>
            <div className={classes.personal}>
                <Heading type="h2">Personal</Heading>
                <Paragraph type="large">
                    Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium
                    dui. In malesuada enim in dolor euismod, id commodo mi
                    consectetur. Curabitur at vestibulum nisi. Nullam vehicula
                    nisi velit. Mauris turpis nisl, molestie ut ipsum et,
                    suscipit vehicula odio.
                </Paragraph>
                <div className={classes.personalImages}>
                    <div className={classes.personalImagesCol}>
                        <img
                            src="https://via.placeholder.com/319x319"
                            alt="figure"
                        />
                        <img
                            src="https://via.placeholder.com/319x178"
                            alt="figure"
                        />
                    </div>
                    <div className={classes.personalImagesCol}>
                        <img
                            src="https://via.placeholder.com/319x178"
                            alt="figure"
                        />
                        <img
                            src="https://via.placeholder.com/319x319"
                            alt="figure"
                        />
                    </div>
                </div>
            </div>
            <div className={classes.contentBlock}>
                <ContentBlock
                    heading="Heading level 1"
                    paragraph="Lorem ipsum dolor sit amet "
                    cta={[{
                        href: '#',
                        children: 'Read more',
                        variant: 'primary'
                    }]}
                />
            </div>
            <ShareSection heading="Follow me" />
            <FooterMock/>
        </div>
    );
};

export default Home;
