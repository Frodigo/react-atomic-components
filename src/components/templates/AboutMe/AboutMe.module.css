
.about {
    overflow: hidden;
}

.content {
    padding: 20px var(--spacing-md);
    text-align: center;
}

.mainFigure {
    composes: figure from '../../organisms/AboutTheAuthor/AboutTheAuthor.module.css';
    margin: 30px 0;
}

.experience {
    border-top: solid 5px var(--color-primary);
    padding-top: 30px;
    text-align: center;
}

.certifications {
    padding: var(--spacing-md);
}

.certificationList {
    margin-top: 30px;
}

.certificationList li {
    color: var(--color-gray-dark-2);
}

.testimonials {
    margin: 30px -15px;
}

.testimonialsHeading {
    border-bottom: solid 5px var(--color-primary);
    margin-left: 30px;
}

.personal {
    padding: var(--spacing-md);
}

.personalImages {
    margin: 30px -15px -15px;
}

.personalImagesCol > div {
    margin-bottom: var(--spacing-md) !important;
}

.contentBlock {
    background-color: #d8d8d8;
    margin: 30px 0 0;
}


@media (min-width: 768px) {
    .content {
        max-width: var(--tablet-container-width);
        margin: 0 auto;
        padding: 50px var(--spacing-md);
    }

    .personalImages {
        display: flex;
        padding: var(--spacing-md) 5px;
    }

    .personalImagesCol {
        flex: 1 1 50%;
        padding: 10px;
    }
}

.mainFigureDesktop {
    composes: mainFigure;
    display: none;
}

@media (min-width: 1024px) {
    .content {
        align-items: center;
        display: flex;
        margin-left: 0;
        text-align: left;
    }

    .mainFigure {
        display: none;
    }

    .mainFigureDesktop {
        display: block;
    }

    .mainContent {
        flex: 1 1 60%;
        padding-left: var(--spacing-xl);
    }

    .experience {
        padding: var(--spacing-md);
        text-align: left;
    }

    .mainParagraph {
        margin-top: 20px;
    }
}

@media (min-width: 1300px) {
    .content,
    .experience,
    .skills,
    .certifications,
    .testimonialsHeading,
    .personal {
        margin: 0 auto;
        max-width: var(--desktop-container-width);
        padding-left: 0;
        padding-right: 0;
    }
}