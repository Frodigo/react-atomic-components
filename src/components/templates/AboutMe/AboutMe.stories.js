import React from 'react';

import { default as AboutMeTemplate } from './AboutMe';

export default {
    title: 'Templates/AboutMe',
    component: AboutMeTemplate
};

const Template = (args) => <AboutMeTemplate {...args} />;

export const AboutMe = Template.bind({});
