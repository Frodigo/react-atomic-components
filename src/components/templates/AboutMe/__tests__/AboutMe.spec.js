import React from 'react';
import { render } from '@testing-library/react';

import AboutMe from '../AboutMe';

describe('AboutMe should be rendered', () => {
    test('AboutMe renders correctly', () => {
        const { container } = render(<AboutMe />);

        expect(container).toMatchSnapshot();
    });
});
