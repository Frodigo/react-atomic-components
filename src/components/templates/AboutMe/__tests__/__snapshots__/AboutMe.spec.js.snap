// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMe should be rendered AboutMe renders correctly 1`] = `
<div>
  <div
    class="about"
  >
    <header
      class="pageHeader"
    >
      <div
        class="container"
      >
        <a
          class="link"
          href="/"
        >
          <img />
        </a>
        <nav
          class="primaryNav"
        >
          <a
            class=" navLink"
            href="#"
          >
            Home
          </a>
          <a
            class="navLinkActive navLink"
            href="#"
          >
            About me
          </a>
          <a
            class=" navLink"
            href="#"
          >
            Blog
          </a>
          <a
            class=" navLink"
            href="#"
          >
            Get in touch
          </a>
        </nav>
      </div>
    </header>
    <div
      class="content"
    >
      <figure
        class="mainFigureDesktop"
        style="background-image: url(null);"
      />
      <section
        class="mainContent"
      >
        <h1
          class=""
        >
          Hey!
        </h1>
        <figure
          class="mainFigure"
          style="background-image: url(null);"
        />
        <p>
          Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. Curabitur at vestibulum nisi. Nullam vehicula nisi velit. Mauris turpis nisl, molestie ut ipsum et, suscipit vehicula odio. Vestibulum interdum vestibulum felis ac molestie.
        </p>
      </section>
    </div>
    <section
      class="experience"
    >
      <h2
        class=""
      >
        Work experience
      </h2>
      <div
        class="mediaBlockSmaller"
      >
        <figure
          class="figure"
        >
          <img
            alt="Frontend Tech Leader"
            src="https://via.placeholder.com/240x240"
          />
        </figure>
        <section
          class="content"
        >
          <span
            class="subtitle"
          >
            05/2016 - 04/2020
          </span>
          <h3
            class=""
          >
            Frontend Tech Leader
          </h3>
          Divante
        </section>
      </div>
      <div
        class="mediaBlockSmaller"
      >
        <figure
          class="figure"
        >
          <img
            alt="Frontend Tech Leader"
            src="https://via.placeholder.com/240x240"
          />
        </figure>
        <section
          class="content"
        >
          <span
            class="subtitle"
          >
            05/2016 - 04/2020
          </span>
          <h3
            class=""
          >
            Frontend Tech Leader
          </h3>
          Divante
        </section>
      </div>
    </section>
    <div
      class="skills"
    >
      <div
        class="carousel"
      >
        <h2
          class="title"
        >
          Tech skills
        </h2>
        <div
          class="slick-slider slick-initialized"
        >
          
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <img
                    alt="figure"
                    height="auto"
                    src="https://via.placeholder.com/208x117"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                    width="auto"
                  />
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active"
                data-index="1"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <img
                    alt="figure"
                    height="auto"
                    src="https://via.placeholder.com/260x117"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                    width="auto"
                  />
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active"
                data-index="2"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <img
                    alt="figure"
                    height="auto"
                    src="https://via.placeholder.com/108x117"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                    width="auto"
                  />
                </div>
              </div>
              <div
                aria-hidden="false"
                class="slick-slide slick-active"
                data-index="3"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <img
                    alt="figure"
                    height="auto"
                    src="https://via.placeholder.com/58x117"
                    style="width: 100%; display: inline-block;"
                    tabindex="-1"
                    width="auto"
                  />
                </div>
              </div>
            </div>
          </div>
          
          
        </div>
      </div>
    </div>
    <div
      class="certifications"
    >
      <h2
        class=""
      >
        Certifications
      </h2>
      <p
        class="paraghaph paragraphLarge "
      >
        Lorem ipsum dolor sit amet Lorem ipsum dolor sit amet Lorem ipsum dolor sit amet Lorem ipsum dolor sit amet Lorem ipsum dolor sit amet
      </p>
      <ul
        class="certificationList"
      >
        <li>
          <a
            href="#"
          >
            Certification 1
          </a>
          (10/2020)
        </li>
        <li>
          <a
            href="#"
          >
            Certification 2
          </a>
          (10/2020)
        </li>
        <li>
          <a
            href="#"
          >
            Certification 3
          </a>
          (10/2020)
        </li>
      </ul>
    </div>
    <div
      class="testimonials"
    >
      <h2
        class="testimonialsHeading"
      >
        Testimonials
      </h2>
      <div
        class="slider"
      >
        <div
          class="slick-slider slick-initialized"
          dir="ltr"
        >
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <figure
                    class="figure"
                  >
                    <div
                      class="inner"
                    >
                      <blockquote
                        class="quote"
                      >
                        <span>
                          svg
                        </span>
                        <p
                          class="paraghaph"
                          style="text-align: center;"
                        >
                          Lorem ipsum
                        </p>
                      </blockquote>
                      <figcaption
                        class="caption"
                      >
                        Author 1
                      </figcaption>
                    </div>
                  </figure>
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide"
                data-index="1"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <figure
                    class="figure"
                  >
                    <div
                      class="inner"
                    >
                      <blockquote
                        class="quote"
                      >
                        <span>
                          svg
                        </span>
                        <p
                          class="paraghaph"
                          style="text-align: center;"
                        >
                          Lorem ipsum
                        </p>
                      </blockquote>
                      <figcaption
                        class="caption"
                      >
                        Author 2
                      </figcaption>
                    </div>
                  </figure>
                </div>
              </div>
              <div
                aria-hidden="true"
                class="slick-slide"
                data-index="2"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <figure
                    class="figure"
                  >
                    <div
                      class="inner"
                    >
                      <blockquote
                        class="quote"
                      >
                        <span>
                          svg
                        </span>
                        <p
                          class="paraghaph"
                          style="text-align: center;"
                        >
                          Lorem ipsum
                        </p>
                      </blockquote>
                      <figcaption
                        class="caption"
                      >
                        Author 3
                      </figcaption>
                    </div>
                  </figure>
                </div>
              </div>
            </div>
          </div>
          <ul
            class="slick-dots"
            style="display: block;"
          >
            <li
              class="slick-active"
            >
              <button>
                1
              </button>
            </li>
            <li
              class=""
            >
              <button>
                2
              </button>
            </li>
            <li
              class=""
            >
              <button>
                3
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="personal"
    >
      <h2
        class=""
      >
        Personal
      </h2>
      <p
        class="paraghaph paragraphLarge "
      >
        Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. Curabitur at vestibulum nisi. Nullam vehicula nisi velit. Mauris turpis nisl, molestie ut ipsum et, suscipit vehicula odio.
      </p>
      <div
        class="personalImages"
      >
        <div
          class="personalImagesCol"
        >
          <img
            alt="figure"
            src="https://via.placeholder.com/319x319"
          />
          <img
            alt="figure"
            src="https://via.placeholder.com/319x178"
          />
        </div>
        <div
          class="personalImagesCol"
        >
          <img
            alt="figure"
            src="https://via.placeholder.com/319x178"
          />
          <img
            alt="figure"
            src="https://via.placeholder.com/319x319"
          />
        </div>
      </div>
    </div>
    <div
      class="contentBlock"
    >
      <section
        class="block"
        style="background-color: transparent;"
      >
        <div
          class="container"
        >
          <h3
            class="heading"
          >
            Heading level 1
          </h3>
          <p
            class="paragraph"
          >
            Lorem ipsum dolor sit amet 
          </p>
          <div
            class="links"
          >
            <a
              class="primary"
              href="#"
            >
              Read more
            </a>
          </div>
        </div>
      </section>
    </div>
    <div
      class="shareSection"
    >
      <h4
        class=""
      >
        Follow me
      </h4>
    </div>
    <form
      action="https://app.getresponse.com/add_subscriber.html"
      class="newsletter"
      id="newsletter"
      method="post"
    >
      <div
        class="newsletterContent"
      >
        <h3
          class="title"
        >
          Subscribe to the newsletter
        </h3>
        <p
          class="paraghaph"
        >
          I will send you unique content every Friday.
        </p>
        <div
          class="fieldset"
        >
          <div
            class="field"
          >
            <input
              class="input null"
              data-testid="input"
              name="first_name"
              placeholder="Your name"
              type="text"
              value=""
            />
          </div>
          <div
            class="field"
          >
            <input
              class="input null"
              data-testid="input"
              name="email"
              placeholder="Your email"
              type="email"
              value=""
            />
          </div>
        </div>
        <input
          name="campaign_token"
          type="hidden"
          value="WMdq9"
        />
        <input
          name="thankyou_url"
          type="hidden"
          value="https://marcin-kwiatkowski.com/thank-you"
        />
        <input
          name="start_day"
          type="hidden"
          value="0"
        />
        <footer
          class="footer"
        >
          <p>
            Also i will send you send you PDF with 112 pages about PWA Studio Rich Content renderers
          </p>
          <button
            class="primary"
            type="submit"
          >
            Subscribe
          </button>
        </footer>
      </div>
    </form>
    <footer
      class="pageFooter"
    >
      <div
        class="columns"
      >
        <div
          class="footerColumn"
        >
          <button
            class="header"
            type="button"
          >
            <h5
              class="heading"
            >
              Column 1
            </h5>
            <span
              class="expandButton"
            >
              <span>
                svg
              </span>
            </span>
          </button>
        </div>
        <div
          class="footerColumn"
        >
          <button
            class="header"
            type="button"
          >
            <h5
              class="heading"
            >
              Column 2
            </h5>
            <span
              class="expandButton"
            >
              <span>
                svg
              </span>
            </span>
          </button>
        </div>
        <div
          class="footerColumn"
        >
          <button
            class="header"
            type="button"
          >
            <h5
              class="heading"
            >
              Column 3
            </h5>
            <span
              class="expandButton"
            >
              <span>
                svg
              </span>
            </span>
          </button>
        </div>
        <div
          class="footerColumn"
        >
          <button
            class="header"
            type="button"
          >
            <h5
              class="heading"
            >
              Column 4
            </h5>
            <span
              class="expandButton"
            >
              <span>
                svg
              </span>
            </span>
          </button>
        </div>
      </div>
      <div>
        <ul
          class="bottomNav"
        >
          <li>
            <a
              href="#"
            >
              Link 1
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Link 2
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Link 3
            </a>
          </li>
        </ul>
        <p
          class="copyright"
        >
          © 2021 Lorem ipsum
        </p>
      </div>
    </footer>
    <nav
      class="mobileNav"
    >
      <a
        href="/"
      >
        <span
          class="iconContainer"
        >
          <img
            alt="Home"
            src="[object Object]"
          />
          <span
            class="label"
          >
            Home
          </span>
        </span>
      </a>
      <a
        href="/blog"
      >
        <span
          class="iconContainer"
        >
          <img
            alt="Blog"
            src="[object Object]"
          />
          <span
            class="label"
          >
            Blog
          </span>
        </span>
      </a>
      <a
        href="/about-me"
      >
        <span
          class="iconContainer"
        >
          <img
            alt="About"
            src="[object Object]"
          />
          <span
            class="label"
          >
            About
          </span>
        </span>
      </a>
      <a
        href="/get-in-touch"
      >
        <span
          class="iconContainer"
        >
          <img
            alt="Contact"
            src="[object Object]"
          />
          <span
            class="label"
          >
            Contact
          </span>
        </span>
      </a>
      <a
        href="#newsletter"
      >
        <span
          class="iconContainer"
        >
          <img
            alt="Subscribe"
            src="[object Object]"
          />
          <span
            class="label"
          >
            Subscribe
          </span>
        </span>
      </a>
    </nav>
  </div>
</div>
`;
